WalletWasabi/Blockchain/Transactions/SmartTransaction.cs

```diff
@@ -126,8 +126,8 @@ public IReadOnlyCollection<WalletVirtualInput> WalletVirtualInputs
 		get
 		{
 			WalletVirtualInputsCache ??= WalletInputs
-					.GroupBy(i => i.HdPubKey.PubKeyHash.ToBytes(), new ByteArrayEqualityComparer())
-					.Select(g => new WalletVirtualInput(g.Key, g.ToHashSet()))
+					.GroupBy(i => i.HdPubKey.PubKey)
+					.Select(g => new WalletVirtualInput(g.Key.ToBytes(), g.ToHashSet()))
```

Comment:
If this is performance critical, you can consider using the unsafe overload of the `ToBytes` method.